### PR TITLE
Input validation bug fix

### PR DESCRIPTION
**Pull request by rjdeloss**
* Requirements *
When a player of TTT has supplied inputs,
the game should not blow up
and the player should be explained why it wasn't good input
So that a player can continue to play even when they give bad input

Fixes input crash. 

When a player enters an input other than the required integer, it will prompt the player to enter a valid value instead of crashing the program.

### CODE REVIEW

File: app/src/main/java/tttTDD/Java/ConsoleClient.java [hunk 1]
```diff
@@ -144,13 +144,26 @@ private String formatCellDisplay(String cell, String maxInput) {
...
     }
+
+    private int playerInput() {
+        int value;
+
+        try {
+            value = input.nextInt();
+        }
+        catch (Exception e) {
+            System.out.println("Stop it. Enter a number as an input:");
```
Comment (joeyshipley) [inline app/src/main/java/tttTDD/Java/ConsoleClient.java:162]: Hahaha.

Comment (rjdeloss) [reply app/src/main/java/tttTDD/Java/ConsoleClient.java:162]: Looool

File: app/src/main/java/tttTDD/Java/ConsoleClient.java [hunk 2]
```diff
@@ -144,13 +144,26 @@ private String formatCellDisplay(String cell, String maxInput) {
...
+            input.nextLine();
+            value = playerInput();
```
Comment (nystromb) [inline app/src/main/java/tttTDD/Java/ConsoleClient.java:164]: What happens if you just `return playerInput();` here?

File: app/src/main/java/tttTDD/Java/ConsoleClient.java [hunk 3]
```diff
@@ -144,13 +144,26 @@ private String formatCellDisplay(String cell, String maxInput) {
...
         }
 
         return formattedCell.toString();
     }
+
+    private int playerInput() {
+        int value;
+
+        try {
+            value = input.nextInt();
```
Comment (nystromb) [inline app/src/main/java/tttTDD/Java/ConsoleClient.java:159]: I wonder if setting the variable here is also necessary. Could you just `return input.nextInt();` here also? 

I think it should still catch the exception and if you re-call this method, it should work? Not positive, but it feels like this _should_ work

File: app/src/main/java/tttTDD/Java/ConsoleClient.java (hunk 2 above)
Comment (rjdeloss) [reply app/src/main/java/tttTDD/Java/ConsoleClient.java:164]: it does the same thing lol. The reason behind using a variable was more for readability purposes.

Comment (nystromb) [reply app/src/main/java/tttTDD/Java/ConsoleClient.java:164]: I think for recursion like this, it kind of added to my confusion. By assigning a variable, at first I thought the code flow would continue on. 

In general, I think having an explicit return here would more clearly indicate the end of the execution path for this function / call stack, when an exception occurs

**Review by nystromb — APPROVED** [review]
LGTM had some comments, but not sure its worth holding up the PR for

**Comment by joeyshipley** [comment]
What happens to the game/feedback-loops if someone supplies a -1 in the board size prompt?

File: app/src/main/java/tttTDD/Java/ConsoleClient.java [hunk 4]
```diff
@@ -144,13 +145,43 @@ private String formatCellDisplay(String cell, String maxInput) {
...
+
+    private int playerInput() {
+        try {
+            int value = input.nextInt();
+
+            if (value > 2) {
+                return value;
+            } else {
+                throw new Exception();
+            }
```
Comment (billwanjohi) [inline app/src/main/java/tttTDD/Java/ConsoleClient.java:164]: Does this not make inappropriately constrain the human player's moves?

Comment (billwanjohi) [reply app/src/main/java/tttTDD/Java/ConsoleClient.java:164]: More generally, I like the idea of a wrapper for the `Scanner` which forces us to handle the non-numeric exception, but it doesn't seem like this shared method should be responsible for handling whether it's the wrong _kind_ of number.

Comment (rjdeloss) [reply app/src/main/java/tttTDD/Java/ConsoleClient.java:164]: After committing it I realized the same thing so I ended making a separate method for the `boardSize` input.

**Pull request merged**